From f1b604d794589125ff2b78a0ee5243e4517561c1 Mon Sep 17 00:00:00 2001
From: Radha Mohan Chintakuntla <radhac@marvell.com>
Date: Thu, 23 Apr 2020 10:37:21 -0700
Subject: [PATCH 1317/1921] misc: mrvl-loki: Add function to unregister the irq
 callback

The function to unregister the irq callback helps in avoiding a kernel
panic if an interrupt arrives and the driver with cabllback is rmmod.

Change-Id: Ia46ac1a67e69bc081026115930f4bd51d1a4cba5
Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/27414
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
(cherry picked from commit 82f31aec767257ef98be94376623e145fe155307)
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/47642
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/misc/mrvl-loki.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/misc/mrvl-loki.c b/drivers/misc/mrvl-loki.c
index 711b26cbd345..c2a444ad0bcf 100644
--- a/drivers/misc/mrvl-loki.c
+++ b/drivers/misc/mrvl-loki.c
@@ -61,6 +61,12 @@ int mrvl_loki_register_irq_cb(connip_irq_cb_t func)
 }
 EXPORT_SYMBOL(mrvl_loki_register_irq_cb);
 
+void mrvl_loki_unregister_irq_cb(void)
+{
+	g_ml->irq_cb = NULL;
+}
+EXPORT_SYMBOL(mrvl_loki_unregister_irq_cb);
+
 static irqreturn_t mrvl_loki_handler(int irq, void *dev)
 {
 	struct mrvl_loki *ml =
-- 
2.31.1

